src/components/main/files/upload/mod.rs

```diff
@@ -18,81 +21,113 @@ pub struct Props<'a> {
     on_hide: EventHandler<'a, MouseEvent>,
 }
 
+enum Action {
+        Start,
+        Stop,
+    }
+
+
+
 #[allow(non_snake_case)]
 pub fn Upload<'a>(cx: Scope<'a, Props<'a>>) -> Element<'a> {
     let file_storage = cx.props.storage.clone();
-
+    let drag_over_dropzone = use_ref(&cx, || false);
+    let eval_script = use_window(&cx).clone();
+    let file_over_dropzone_js = include_str!("./file_over_dropzone.js");
+    let file_leave_dropzone_js = include_str!("./file_leave_dropzone.js");
+    let file_being_uploaded_js = "document.getElementById('dropzone').value = 'Uploading...'";
+
+
+    let upload_file_dropped_routine = use_coroutine(&cx, |mut rx: UnboundedReceiver<Action>| {
+        to_owned![file_storage, drag_over_dropzone, eval_script, file_leave_dropzone_js, file_over_dropzone_js, file_being_uploaded_js];
+        async move {
+        while let Some(action) = rx.next().await {
+            match action {
+                Action::Start => {
+                            log::info!("File on dropzone");
+                            tokio::time::sleep(Duration::from_millis(100)).await;
+                        if *drag_over_dropzone.read() {
+                            let dropped_file = DROPPED_FILE.read();
+                            // TODO(use_eval): Try new solution in the future
+                            if dropped_file.files_local_path.len() > 1 {
+                                let files_to_upload = format!("{} files to upload!", dropped_file.files_local_path.len());
+                                eval_script.eval(&file_over_dropzone_js.replace("file_path", &files_to_upload));
+                            } else {
+                                eval_script.eval(&file_over_dropzone_js.replace("file_path", &dropped_file.files_local_path[0]));
+                            }
+                            if dropped_file.file_drag_event == FileDragEvent::Dropped {
+                                *drag_over_dropzone.write_silent() = false;
+                                  // TODO(use_eval): Try new solution in the future
+                                  eval_script.eval(&file_being_uploaded_js);
+                                for file_path in &dropped_file.files_local_path {
+                                    println!("file path: {:?}", file_path);
+                                    let file_path_buf = std::path::Path::new(&file_path.trim()).to_path_buf();     
+                                    upload_file(file_storage.clone(), file_path_buf).await;
```

Comment:
This might be a good example of where one could use `Constellation::put_stream`. Reasoning for that is youre dealing with multiple files which can vary on different sizes and `Constellation::put` would yield until it is uploaded while `Constellation::put_stream` would return `ConstellationProgressStream` that one could then use to get the progress of each one being uploaded. Change is not required right now though as there would need to be a way to track the task monitoring it.